### PR TITLE
MULE-11856: @Xml annotation attributes renaming

### PR DESCRIPTION
**Pull request by fernandezlautaro**
depends on https://github.com/mulesoft/mule-api/pull/162 and https://github.com/mulesoft/mule-extensions-api/pull/269

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/connection/ErrorTypeHandlerConnectionProviderWrapper.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public ErrorTypeHandlerConnectionProviderWrapper(ConnectionProvider<C> connectio
                                                    ExtensionModel extensionModel, RetryPolicyTemplate retryPolicyTemplate) {
     super(connectionProvider);
     this.errorTypeRepository = muleContext.getErrorTypeRepository();
-    namespace = extensionModel.getXmlDslModel().getNamespace().toUpperCase();
+    namespace = extensionModel.getXmlDslModel().getPrefix().toUpperCase();
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/internal/connection/ErrorTypeHandlerConnectionProviderWrapper.java]: rename variable to prefix

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/manager/DefaultExtensionManagerTestCase.java [hunk 2]
```diff
@@ -80,7 +80,7 @@
 
   private static final String MULESOFT = "MuleSoft";
   private static final String OTHER_VENDOR = "OtherVendor";
-  private static final XmlDslModel XML_DSL_MODEL = XmlDslModel.builder().setNamespace("extension-namespace").build();
+  private static final XmlDslModel XML_DSL_MODEL = XmlDslModel.builder().setPrefix("extension-namespace").build();
```
Comment (marianogonzalez) [inline modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/manager/DefaultExtensionManagerTestCase.java]: `extension-namespace` -> `extension-prefix`

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/manager/ExtensionErrorsRegistrantTestCase.java [hunk 3]
```diff
@@ -96,7 +96,7 @@
   @Before
   public void setUp() {
     XmlDslModel.XmlDslModelBuilder builder = XmlDslModel.builder();
-    builder.setNamespace(EXTENSION_NAMESPACE);
+    builder.setPrefix(EXTENSION_NAMESPACE);
```
Comment (marianogonzalez) [inline modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/manager/ExtensionErrorsRegistrantTestCase.java]: rename constant

**Comment by fernandezlautaro** [comment]
ran the tests in the following folders
 _ mule/core/
 _ mule/core-tests/
 _ mule/modules/extensions-xml-support
 _ mule/modules/extensions-spring-support
 _ mule/modules/extensions-support
 _ mule/modules/dsl-api
 _ mule/modules/spring-config
 _ mule/extensions/http
 _ mule/modules/http
 _ mule/tests/dsl
 _ mule/tests/integration

File: core/src/main/java/org/mule/runtime/core/internal/connection/ErrorTypeHandlerConnectionProviderWrapper.java (hunk 1 above)
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/internal/connection/ErrorTypeHandlerConnectionProviderWrapper.java]: ignored comment

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/manager/DefaultExtensionManagerTestCase.java (hunk 2 above)
Comment (marianogonzalez) [inline modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/manager/DefaultExtensionManagerTestCase.java]: ignored comment

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/manager/ExtensionErrorsRegistrantTestCase.java (hunk 3 above)
Comment (marianogonzalez) [inline modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/manager/ExtensionErrorsRegistrantTestCase.java]: ignored comment

File: core/src/main/java/org/mule/runtime/core/internal/connection/ErrorTypeHandlerConnectionProviderWrapper.java (hunk 1 above)
Comment (fernandezlautaro) [inline core/src/main/java/org/mule/runtime/core/internal/connection/ErrorTypeHandlerConnectionProviderWrapper.java]: not sure if you didn't see the commit [fixes after MGs review](https://github.com/mulesoft/mule/pull/4180/commits/01c58861231d7e49389197e63138a30452b2d9a2), but they weren't ignored

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/manager/DefaultExtensionManagerTestCase.java (hunk 2 above)
Comment (fernandezlautaro) [inline modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/manager/DefaultExtensionManagerTestCase.java]: not sure if you didn't see the commit [fixes after MGs review](https://github.com/mulesoft/mule/pull/4180/commits/01c58861231d7e49389197e63138a30452b2d9a2), but they weren't ignored

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/manager/ExtensionErrorsRegistrantTestCase.java (hunk 3 above)
Comment (fernandezlautaro) [inline modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/manager/ExtensionErrorsRegistrantTestCase.java]: not sure if you didn't see the commit [fixes after MGs review](https://github.com/mulesoft/mule/pull/4180/commits/01c58861231d7e49389197e63138a30452b2d9a2), but they weren't ignored

**Pull request merged**